util/HalideTraceUtils.h

```diff
@@ -74,7 +74,11 @@ struct Packet : public halide_trace_packet_t {
     template<typename T>
     T get_value_as(int idx) const {
         const uint8_t *val = (const uint8_t *)(value()) + idx * type.bytes();
-        return value_as<T>(type, (const halide_scalar_value_t *)val);
+        // 'val' may not be aligned: memcpy it to an aligned local
+        // so that value_as<>() won't complain under sanitizers.
+        halide_scalar_value_t aligned_value;
+        memcpy(&aligned_value, val, sizeof(aligned_value));
```

Comment:
Is this the right size? This seems like it will copy the largest sized type worth of bytes (and possibly more if there is padding at the end of the struct) from the packet. If the value can be unaligned in the packet, I don't see why the packet is guaranteed to be valid for that number of bytes.

This code also assumes little endian. If one is going to do that, then just copying the number of bytes for the type is ok.

Reply:
Ah, good point -- the runtime code that generates this bytestream doesn't guarantee that size. Will rework.